tenable/ad/widget/api.py

```diff
@@ -0,0 +1,260 @@
+        payload = self._schema.dump(self._schema.load({
+            'posX': pos_x,
+            'posY': pos_y,
+            'width': width,
+            'height': height,
+            'title': title
+        }))
+
+        return self._schema.load(self._post(f'{dashboard_id}/widgets', json=payload))
+
+    def details(self,
+                dashboard_id: str,
+                widget_id: int) -> Dict:
+        '''
+        Retrieves the details for a specific widget.
+
+        Args:
+            dashboard_id: The dashboard instance identifier.
+            widget_id: The widget instance identifier
+
+        Returns:
+            dict:
+                the widget object.
+
+        Examples:
+
+            >>> tad.widget.details(1)
+        '''
+        return self._schema.load(self._get(f"{dashboard_id}/widgets/{widget_id}"))
+
+    def update(self,
+               dashboard_id: str,
+               widget_id: int,
+               **kwargs) -> Dict:
+        '''
+        update an existing widget
+
+        Args:
+            dashboard_id:
+                The dashboard instance identifier.
+            widget_id:
+                The dashboard instance identifier.
+            pos_x:
+                x-axis position for widget.
+            pos_y:
+                y-axis position for widget.
+            width:
+                width of widget.
+            height:
+                height of widget.
+            title:
+                title for widget.
+
+
+        Returns:
+            dict:
+                The updated widget object.
+
+        Examples:
+
+            >>> tad.widgets.update(
+            ...     dashboard_id=1,
+            ...     widget_id=1,
+            ...     pos_x=1,
+            ...     pos_y=1,
+            ...     width=3,
+            ...     height=3,
+            ...     title='EditedWidget'
+            ...     )
+        '''
+        payload = self._schema.dump(kwargs)
+        return self._schema.load(self._patch(f"{dashboard_id}/widgets/{widget_id}", json=payload))
+
+    def delete(self,
+               dashboard_id: str,
+               widget_id: int) -> None:
+        '''
+        delete an existing widget
+
+        Args:
+            dashboard_id:
+                The dashboard instance identifier.
+            widget_id:
+                The widget instance identifier.
+
+
+        Returns:
+            None:
+
+        Examples:
+
+            >>> tad.widgets.delete(
+            ...     dashboard_id=1,
+            ...     widget_id=1
+            ...     )
+        '''
+        self._delete(f"{dashboard_id}/widgets/{widget_id}")
+
+    def widget_options_details(self,
+                               dashboard_id: str,
+                               widget_id: int) -> Dict:
+        '''
+        get widget options
+
+        Args:
+            dashboard_id:
+                The dashboard instance identifier.
+            widget_id:
+                The dashboard instance identifier.
+
```

Comment:
superfluous carriage return

Reply:
This has been resolved in latest commit.